lib/pages/settings_pages/settings_page.dart

```diff
@@ -146,10 +147,9 @@ class _SettingsPageState extends State<SettingsPage> {
               height: 10,
             ),
             SettingsMenuWidget(
-                enableTracking!
-                    ? MyLocalizations.of(context, 'enable_background_tracking')
-                    : MyLocalizations.of(
-                        context, 'disable_background_tracking'), () {
+                isBgTrackingEnabled
+                    ? MyLocalizations.of(context, 'disable_background_tracking')
+                    : MyLocalizations.of(context, 'enable_background_tracking'), () {
```

Comment:
How would you see replacing this 'dialog box' with a toggle switch, continuing with permissions management as it is done now?

Reply:
sure, I can take a look into adding a toggle switch

Reply:
Great! To make things easier, let's use the `Colors.orange` color when the switch is active. Also, I think a toggle switch would work better for boolean options. By the way, we could save some time by reusing the Material implementation of the switch.